P0 - Alert and avatar components

bharati-21 opened the pull request:
- Created and styled alert component. Added it to the docs page.
  - The alert component has 2 variations - simple alerts and simple alerts with a close icon.
  - Used JavaScript for close alert functionality. 
- Created and styled avatar component. Added it to the docs page.
  - The avatar component has 4 variations - circle image avatar, square image avatar, circle text avatar, and square text avatar

bharati-21 committed:
styles: alerts and variations

Styled alerts with variations - simple and with close icon

bharati-21 committed:
feat: close alert on close icon click

bharati-21 committed:
feat: added alerts to doc page

bharati-21 committed:
refactor: rename alert script file

bharati-21 committed:
style: avatar component

Added 4 types of avatar component - round image, square image, round text and square text

bharati-21 committed:
styles: alert width and work wrap

pnchinmay commented:
Hey @bharati-21, the preview link isn't working bcoz `./Components/alerts/index.html` isn't linked in `./index.html` present in root folder.
Also, I would suggest to name it as `./Components/alerts/alert.html`

![image](https://user-images.githubusercontent.com/62750974/151829455-14d85ac3-18f1-41bd-a285-3c2684b23725.png)
Margin isn't present on the right side. Also, keep the width of these alerts max upto half width of the page.
**Colors are awesome.** Those close buttons are working great

pnchinmay commented:
Avatars look great

bharati-21 commented:
> Hey @bharati-21, the preview link isn't working bcoz `./Components/alerts/index.html` isn't linked in `./index.html` present in root folder. Also, I would suggest to name it as `./Components/alerts/alert.html`
> 
> ![image](https://user-images.githubusercontent.com/62750974/151829455-14d85ac3-18f1-41bd-a285-3c2684b23725.png) Margin isn't present on the right side. Also, keep the width of these alerts max upto half width of the page. **Colors are awesome.** Those close buttons are working great.


----

Hey, @pnchinmay  thank you for your review.

Actually, the docs page is not the one that's previewed by netlify here. It's the landing page (which I'm yet to add). 
You can look at the docs page here https://deploy-preview-1--elixir-ui.netlify.app/docs/ . I had mentioned it when I dropped the PR request.

As for the `alert/index.html`, it was just a page I added to visualize my alerts outside of docs. But I shall add margin and keep width up to half the width.

Thanks. Please do check the docs page once https://deploy-preview-1--elixir-ui.netlify.app/docs/ so that I can understand what to remove/ add to the docs.

pnchinmay commented:
> Actually, the docs page is not the one that's previewed by netlify here. It's the landing page (which I'm yet to add). You can look at the docs page here https://deploy-preview-1--elixir-ui.netlify.app/docs/ . I had mentioned it when I dropped the PR request.
> 
> As for the `alert/index.html`, it was just a page I added to visualize my alerts outside of docs. But I shall add margin and keep width up to half the width.
> 
> Thanks. Please do check the docs page once https://deploy-preview-1--elixir-ui.netlify.app/docs/ so that I can understand what to remove/ add to the docs.

Sorry about that. I should've read the msg.
Docs site looks great. Good to go

bharati-21 commented:
> > Actually, the docs page is not the one that's previewed by netlify here. It's the landing page (which I'm yet to add). You can look at the docs page here https://deploy-preview-1--elixir-ui.netlify.app/docs/ . I had mentioned it when I dropped the PR request.
> > As for the `alert/index.html`, it was just a page I added to visualize my alerts outside of docs. But I shall add margin and keep width up to half the width.
> > Thanks. Please do check the docs page once https://deploy-preview-1--elixir-ui.netlify.app/docs/ so that I can understand what to remove/ add to the docs.
> 
> Sorry about that. I should've read the msg. Docs site looks great. Good to go

No problem!
Thank you for your time and review! :D

yashpatel1797 commented on Components/alerts/alert.css:
```diff
@@ -0,0 +1,92 @@
+:root {
+    --secondary-bg-color: #c3dbff;
+    --warning-bg-color: #fce6cf;
+    --error-bg-color: #ffd0d0;
+    --success-bg-color: #a8fcd6;
+}
```
If this is common colour then you can include it in base CSS but if specific to only alert then fine.

bharati-21 replied:
This is specific to alert. In the future, if it becomes common I shall move it. 👍🏽

yashpatel1797 commented on Components/alerts/index.html:
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
As per Smile Gupta PR review, you can add a name in the meta tag.

bharati-21 replied:
Yes, I shall add it 👍🏽

bharati-21 replied:
Added desc and author meta tags.

yashpatel1797 commented on docs/index.html:
```diff
@@ -69,9 +69,256 @@ <h4 class="sidebar__head">
+                            <div class="code-display">
+                                <iframe
+                                    src="https://carbon.now.sh/embed/uUNSWLE4IDf7K8vCVq6y"
+                                    style="width: 100%;"
```
Avoid writing inline CSS.

bharati-21 replied:
Yes, will remove this.

yashpatel1797 reviewed (commented):
1- Rename avatars/index.html to avatars/avatar.html. 
2- The naming convention is good.
3- Design is clear and colour choice is also good.

bharati-21 commented:
> 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.

1. The index.html file is one I created for visualizing the components alone, outside of the component library. I'm not using it on the website. But yes, I shall change the name or remove it, since it's causing a lot of confusion.
2. Thank you!
3. Thank you!

bharati-21 committed:
refactor: add author, desc to docs page

bharati-21 committed:
refactor: add author, desc to home

bharati-21 committed:
refactor: edit docs page path in home

bharati-21 committed:
refactor: edit docs page title

bharati-21 committed:
styles: remove iframe inline styles

bharati-21 commented:
> 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.

@yashpatel1797, I made the changes you asked for. Can you please check them once and approve them?
Thanks.

yashpatel1797 commented:
> > 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.
> 
> @yashpatel1797, I made the changes you asked for. Can you please check them once and approve them? Thanks.



> > 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.
> 
> 1. The index.html file is one I created for visualizing the components alone, outside of the component library. I'm not using it on the website. But yes, I shall change the name or remove it, since it's causing a lot of confusion.
> 2. Thank you!
> 3. Thank you!

@bharati-21
1> Actually I see that you have mentioned that but I haven't read it, a mistake from my side.

bharati-21 commented:
> > > 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.
> > 
> > 
> > @yashpatel1797, I made the changes you asked for. Can you please check them once and approve them? Thanks.
> 
> > > 1- Rename avatars/index.html to avatars/avatar.html. 2- The naming convention is good. 3- Design is clear and colour choice is also good.
> > 
> > 
> > 
> > 1. The index.html file is one I created for visualizing the components alone, outside of the component library. I'm not using it on the website. But yes, I shall change the name or remove it, since it's causing a lot of confusion.
> > 2. Thank you!
> > 3. Thank you!
> 
> @bharati-21 1> Actually I see that you have mentioned that but I haven't read it, a mistake from my side.

No problem! 
Thank you for your time and review 😄
